api/hypershift/v1beta1/azure_types.go

```diff
@@ -0,0 +1,213 @@
+	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
+	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
+	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
+	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
+	// +kubebuilder:validation:Required
+	Image AzureVMImage `json:"image"`
+
+	// diskSizeGiB is the size in GiB (1024^3 bytes) to assign to the OS disk. This should be between 16 and 65,536.
+	// When not set, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is 30.
+	//
+	// +kubebuilder:validation:Minimum=16
+	// +kubebuilder:validation:Maximum=65536
+	// +optional
+	DiskSizeGB int32 `json:"diskSizeGB,omitempty"`
+
+	// diskStorageAccountType is the disk storage account type to use.
+	// Valid values are Standard, StandardSSD, PremiumSSD and UltraSSD and omitted.
+	// Note that Standard means a HDD.
+	// The disk performance is tied to the disk type, please refer to the Azure documentation for further details
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison.
+	// When omitted this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is PremiumSSD.
+	//
+	// +kubebuilder:validation:Enum=Standard;StandardSSD;PremiumSSD;UltraSSD
+	// +optional
+	// TODO: Should all disk options come under a struct rather than be top level like this?
+	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
+
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Example values are TODO
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// diskEncryptionSetID is the ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs. This
+	// needs to exist in the same subscription id listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.SubscriptionID.
+	// DiskEncryptionSetID should also exist in a resource group under the same subscription id and the same location
+	// listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.Location.
+	// TODO: What does this do for a customer, why would they want to set it?
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
```

Comment:
"Supported characters for the disk encryption set name are letters, numbers, underscores, hyphens, and periods. The name cannot begin with an underscore, hyphen, or period, and cannot end with a hyphen or period.

The value must be between 1 and 80 characters long."

Reply:
When you say the actual name, is that not the value of this field?